### PR TITLE
Created new tests and added more username and password checking on signup

### PR DESCRIPTION
**Pull request by LatinScribe**
- Added username and password validation for user signup
- Added more tests (pretty much all of my code should now be covered)
- Deleted/commented out some old code

### CODE REVIEW

**Comment by LatinScribe** [comment]
Also fixed the backspace bug

File: src/main/java/app/LoginUseCaseFactory.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public static LoginView create(
             LoginController loginController = createLoginUseCase(viewManagerModel, loginViewModel, loggedInViewModel, mainMenuViewModel, userDataAccessObject, tokenDataAccessInterface);
             return new LoginView(loginViewModel, loginController);
         } catch (IOException e) {
-            JOptionPane.showMessageDialog(null, "Could not open user data file.");
+            JOptionPane.showMessageDialog(null, "Error creating login model. This could be because there was a user DAO issue");
```
Comment (aclin05) [inline src/main/java/app/LoginUseCaseFactory.java:38]: Good added detail on the failure message!

File: src/main/java/app/Main.java [hunk 2]
```diff
@@ -96,9 +92,11 @@ public static void main(String[] args) {
         FileTokenDataAccessObject tokenDataAccessObject = new FileTokenDataAccessObject();
         UserPreferencesFactory userPreferencesFactory = new UserPreferencesFactory();
         ServerProfileDataAccessObject profileDataAccessObject = new ServerProfileDataAccessObject(tokenDataAccessObject, userPreferencesFactory);
+        PasswordValidatorService passwordValidatorService = new PasswordValidatorService();
```
Comment (aclin05) [inline src/main/java/app/Main.java:95]: Implementation of the Validator Services allow for a good implementation of the single responsibility principle (SOLID)!

File: src/main/java/entity/PasswordValidator.java [hunk 3]
```diff
@@ -3,4 +3,5 @@
 public interface PasswordValidator {
     public boolean passwordIsValid(String password);
 
+    String getRule();
```
Comment (aclin05) [inline src/main/java/entity/PasswordValidator.java:6]: Validator service seems to work as intended (through testing a use case of less than 5 characters)!

File: src/main/java/entity/UsernameValidatorService.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package entity;
+
+public class UsernameValidatorService implements UsernameValidator{
```
Comment (aclin05) [inline src/main/java/entity/UsernameValidatorService.java:3]: After testing this code, the UsernameValidator seems to work as intended also, with a test done on a 1 character username being successful.

**Review by aclin05 — APPROVED** [review]
Everything seems to be working fine in this use case! Good code cleanup and use of a SOLID principle!

**Pull request merged**